### PR TITLE
Lazy load large activity details in Chat View

### PR DESCRIPTION
**Pull request by is0692vs**
This PR refactors the Chat View to implement lazy loading for large \`plan\`, \`diff\`, and \`artifact\` details to avoid upfront HTML generation and improve performance.

Changes include:
- Generating placeholder HTML for large details with loading text in \`src/chatView.ts\`.
- Handling \`requestDetails\` messages in \`JulesChatViewProvider\` to generate and return details HTML on demand.
- Modifying \`src/webview/chatAssets.ts\` to cache \`detailsHtml\`, listen to \`<details>\` toggle events for lazy loading triggers, and dynamically update the DOM with received HTML content.
- Verified functionality via test suites to ensure no regressions.

---
*PR created automatically by Jules for task [9186281186082688297](https://jules.google.com/task/9186281186082688297) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

Chat View の `plan`・`diff`・`artifact` details をオンデマンドで生成する遅延ロード機構を実装するPRです。プレースホルダー HTML の生成（`chatView.ts`）、`requestDetails` / `detailsHtml` メッセージによるホスト↔webview 間の通信、キャッシュ管理（`chatAssets.ts`）の3層で構成されています。

- **P1**: `renderMessages()` が `chatState` 更新のたびに DOM を全再構築しますが、`expandedDetails` を参照して開いていたパネルを復元するコードがないため、メッセージが届くたびに展開済みパネルが閉じてしまいます。
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

P1 バグ（状態更新のたびにパネルが折りたたまれる）があり、マージ前に修正が推奨されます。

P1 が1件あるため上限 4/5。セキュリティや破壊的な問題はなく、実装の方向性は正しい。expandedDetails を renderMessages() に組み込む修正は局所的で低リスクです。

src/webview/chatAssets.ts の renderMessages() 関数に注目してください。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/chatView.ts | 大きな details を遅延ロード用プレースホルダーに置き換え、handleRequestDetails を追加して要求時に HTML を生成・送信する実装。ロジックは概ね正しい。 |
| src/webview/chatAssets.ts | detailsCache と expandedDetails を追加し toggle/message ハンドラーを実装。ただし renderMessages() が expandedDetails を使って展開状態を復元しておらず、状態更新のたびにパネルが閉じる P1 バグがある。 |
| src/test/chatView.unit.test.ts | 遅延ロードのプレースホルダー生成と handleRequestDetails の各パスを網羅するユニットテストを追加。エッジケース（無効 index、BigInt、unknown type）もカバーしている。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant W as Webview (chatAssets.ts)
    participant E as Extension (chatView.ts)

    E->>W: postMessage(chatState) renderMessages() → innerHTML 再構築
    Note over W: 全 details は Loading... で閉じた状態

    W->>W: ユーザーが details を開く (toggle)
    W->>W: expandedDetails.add(key)
    alt detailsCache[key] なし
        W->>E: postMessage(requestDetails)
        E->>E: handleRequestDetails() renderChatMarkdown() で HTML 生成
        E->>W: postMessage(detailsHtml)
        W->>W: detailsCache[key] = html
        W->>W: DOM の .details-content を更新
    end

    E->>W: postMessage(chatState) 新メッセージ到着
    Note over W: renderMessages() が innerHTML を再構築
    Note over W: expandedDetails は残っているが復元されない P1 バグ
```

<details open><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `src/webview/chatAssets.ts`, line 77-87 ([link](https://github.com/hiroki-org/jules-extension/blob/c57e17658151ac309894c36ae43e4077ecfcb35b/src/webview/chatAssets.ts#L77-L87)) 

   <a href="#">P1</a> **renderMessages() が expandedDetails の状態を復元しない**

   `renderMessages()` は `chatContainer.innerHTML` を毎回完全に上書きするため、それ以前に展開されていた `<details>` 要素はすべて閉じた状態に戻ります。`expandedDetails` セットはトグルハンドラーで維持されていますが、`renderMessages()` 内でその内容を参照してパネルを再展開するコードが存在しないため、`chatState` メッセージが届くたびに（新しいメッセージの到着ごとに発生）展開済みパネルが折りたたまれてしまいます。

   `renderMessages()` 内で DOM 再構築後に `expandedDetails` を使ってパネルを復元する必要があります。例:

   ```js
   function renderMessages() {
     chatContainer.innerHTML = state.messages.map(m => `...`).join("");
     // 展開状態を復元する
     expandedDetails.forEach(key => {
       const [activityId, detailType, indexStr] = key.split("|");
       for (const el of chatContainer.querySelectorAll(".activity-details")) {
         if (
           el.getAttribute("data-activity-id") === activityId &&
           el.getAttribute("data-detail-type") === detailType &&
           (el.getAttribute("data-index") || "") === indexStr
         ) {
           el.open = true;
           const contentDiv = el.querySelector(".details-content");
           if (contentDiv && detailsCache[key]) {
             contentDiv.innerHTML = detailsCache[key];
           }
           break;
         }
       }
     });
     typingIndicator.classList.toggle("visible", !!state.isTyping);
     chatContainer.scrollTop = chatContainer.scrollHeight;
     updateUI();
   }
   ```

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: src/webview/chatAssets.ts
   Line: 77-87

   Comment:
   **renderMessages() が expandedDetails の状態を復元しない**

   `renderMessages()` は `chatContainer.innerHTML` を毎回完全に上書きするため、それ以前に展開されていた `<details>` 要素はすべて閉じた状態に戻ります。`expandedDetails` セットはトグルハンドラーで維持されていますが、`renderMessages()` 内でその内容を参照してパネルを再展開するコードが存在しないため、`chatState` メッセージが届くたびに（新しいメッセージの到着ごとに発生）展開済みパネルが折りたたまれてしまいます。

   `renderMessages()` 内で DOM 再構築後に `expandedDetails` を使ってパネルを復元する必要があります。例:

   ```js
   function renderMessages() {
     chatContainer.innerHTML = state.messages.map(m => `...`).join("");
     // 展開状態を復元する
     expandedDetails.forEach(key => {
       const [activityId, detailType, indexStr] = key.split("|");
       for (const el of chatContainer.querySelectorAll(".activity-details")) {
         if (
           el.getAttribute("data-activity-id") === activityId &&
           el.getAttribute("data-detail-type") === detailType &&
           (el.getAttribute("data-index") || "") === indexStr
         ) {
           el.open = true;
           const contentDiv = el.querySelector(".details-content");
           if (contentDiv && detailsCache[key]) {
             contentDiv.innerHTML = detailsCache[key];
           }
           break;
         }
       }
     });
     typingIndicator.classList.toggle("visible", !!state.isTyping);
     chatContainer.scrollTop = chatContainer.scrollHeight;
     updateUI();
   }
   ```

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
src/webview/chatAssets.ts:77-87
**renderMessages() が expandedDetails の状態を復元しない**

`renderMessages()` は `chatContainer.innerHTML` を毎回完全に上書きするため、それ以前に展開されていた `<details>` 要素はすべて閉じた状態に戻ります。`expandedDetails` セットはトグルハンドラーで維持されていますが、`renderMessages()` 内でその内容を参照してパネルを再展開するコードが存在しないため、`chatState` メッセージが届くたびに（新しいメッセージの到着ごとに発生）展開済みパネルが折りたたまれてしまいます。

`renderMessages()` 内で DOM 再構築後に `expandedDetails` を使ってパネルを復元する必要があります。例:

```js
function renderMessages() {
  chatContainer.innerHTML = state.messages.map(m => `...`).join("");
  // 展開状態を復元する
  expandedDetails.forEach(key => {
    const [activityId, detailType, indexStr] = key.split("|");
    for (const el of chatContainer.querySelectorAll(".activity-details")) {
      if (
        el.getAttribute("data-activity-id") === activityId &&
        el.getAttribute("data-detail-type") === detailType &&
        (el.getAttribute("data-index") || "") === indexStr
      ) {
        el.open = true;
        const contentDiv = el.querySelector(".details-content");
        if (contentDiv && detailsCache[key]) {
          contentDiv.innerHTML = detailsCache[key];
        }
        break;
      }
    }
  });
  typingIndicator.classList.toggle("visible", !!state.isTyping);
  chatContainer.scrollTop = chatContainer.scrollHeight;
  updateUI();
}
```


`````

</details>

<sub>Reviews (3): Last reviewed commit: ["Add fixed edge case tests for lazy loadi..."](https://github.com/hiroki-org/jules-extension/commit/c57e17658151ac309894c36ae43e4077ecfcb35b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30153237)</sub>

### CODE REVIEW

File: src/webview/chatAssets.ts [hunk 1]
```diff
@@ -107,6 +109,29 @@ export const CHAT_JS = `(function() {
...
+      if (activityId && detailType) {
+        const indexStr = details.getAttribute("data-index");
+        const index = indexStr ? parseInt(indexStr, 10) : undefined;
+        const key = activityId + "|" + detailType + "|" + (indexStr || "");
+
+        if (details.open) {
+          expandedDetails.add(key);
+          if (!detailsCache[key]) {
+            vscode.postMessage({ type: "requestDetails", activityId, detailType, index });
+          }
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:127]: 対応しました: キャッシュヒット時も `.details-content` に `detailsCache` の HTML を再適用するようにしました。影響範囲: lazy-loaded details の toggle 処理。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/webview/chatAssets.ts [hunk 2]
```diff
@@ -126,6 +151,23 @@ export const CHAT_JS = `(function() {
...
+      detailsEls.forEach(details => {
+        const elIndex = details.getAttribute("data-index") || "";
+        const msgIndex = index !== undefined ? String(index) : "";
+        if (elIndex === msgIndex) {
+          const contentDiv = details.querySelector(".details-content");
+          if (contentDiv) {
+            contentDiv.innerHTML = html;
+          }
+        }
+      });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:170]: 対応しました: `activityId` / `detailType` を CSS selector 文字列へ直接補間する処理をやめ、`.activity-details` を列挙して `getAttribute()` で照合する実装に変更しました。影響範囲: `detailsHtml` 受信時の DOM 更新。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/webview/chatAssets.ts [hunk 3]
```diff
@@ -59,6 +59,8 @@ export const CHAT_JS = `(function() {
   const sessionLabel = document.getElementById("sessionLabel");
 
   let state = { sessionId: null, messages: [], isTyping: false };
+  let detailsCache = {}; // "activityId|detailType|index" -> html
+  let expandedDetails = new Set(); // set of "activityId|detailType|index"
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:63]: 対応しました: `expandedDetails` を再描画後の状態復元に使うようにし、`expandedDetails.has(key)` で `<details>` の open 状態を復元します。影響範囲: lazy-loaded details の rerender 復元処理。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/webview/chatAssets.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:170]: 対応しました: `chatState` 再描画後に現在レンダリングされている details key を基準に `detailsCache` / `expandedDetails` を prune し、消えた activity の古い HTML を保持しないようにしました。既存 activity のキャッシュは UX 維持のため再適用します。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/webview/chatAssets.ts [hunk 4]
```diff
@@ -126,6 +151,23 @@ export const CHAT_JS = `(function() {
     if (e.data.type === "chatState") {
       state = e.data.payload || state;
       renderMessages();
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:153]: 対応しました: `renderMessages()` 後に `syncRenderedDetailsState()` を呼び、キャッシュ済み HTML と展開状態を再適用するようにしました。あわせて webview HTML の主要 `aria-label` 検証を追加しました。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/webview/chatAssets.ts [hunk 5]
```diff
@@ -126,6 +151,23 @@ export const CHAT_JS = `(function() {
...
+    } else if (e.data.type === "detailsHtml") {
+      const { activityId, detailType, index, html } = e.data;
+      const key = activityId + "|" + detailType + "|" + (index !== undefined ? index : "");
+      detailsCache[key] = html;
+
+      // Update DOM if currently rendered
+      const detailsEls = chatContainer.querySelectorAll('[data-activity-id="' + activityId + '"][data-detail-type="' + detailType + '"]');
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/webview/chatAssets.ts:160]: 対応しました: raw attribute 値を使った CSS selector 生成を削除し、`getAttribute()` ベースの照合に変更しました。`activityId` などに selector 特殊文字が含まれても `querySelectorAll` の構文エラーにならない形です。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

File: src/chatView.ts [hunk 6]
```diff
@@ -381,6 +361,74 @@ export class JulesChatViewProvider implements vscode.WebviewViewProvider {
     };
     this.postState();
   }
+  private handleRequestDetails(message: any): void {
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply src/chatView.ts:364]: 対応しました: `handleRequestDetails` の引数を `RequestDetailsMessage` interface に変更し、`activityId` / `detailType` / `index` を型ガードしてから使うようにしました。影響範囲: requestDetails payload handling。検証: `pnpm run check-types`, `pnpm run lint`, `pnpm run test:unit`。

**Comment by is0692vs** [comment]
レビュー指摘対応と thread resolve を完了しました。CI/コンフリクト確認済みです。再レビューお願いします。

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@coderabbitai review

**Comment by is0692vs** [comment]
@greptile
